sql/core/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -726,6 +726,14 @@ private[sql] class SQLConf extends Serializable with CatalystConf with Logging {
   }
 
   /**
+   * Return the value of an optional Spark SQL configuration property for the given key. If the key
+   * is not set yet, returns None.
+   */
+  def getOptionalConf[T](entry: OptionalConfigEntry[T]): Option[T] = {
```

Comment:
Is `def getConf[T](entry: OptionalConfigEntry[T]): T` (right above this one) used anymore? If not, perhaps it could just be changed to return `Option[T]` instead of doing a `getOrElse`?

If for some reason it's still used, you need the `require(...)` check in this method too, to be consistent with other methods in this class.